core/src/main/scala/chisel3/Bits.scala

```diff
@@ -1708,11 +1704,10 @@ package experimental {
     extends Bits(range.getWidth) with Num[Interval] with HasBinaryPoint {
 
     override def toString: String = {
-      val bindingString = litOption match {
-        case Some(value) => s"($value)"
-        case _ => bindingToString
+      litOption match {
+        case Some(value) => stringAccessor(Some("Interval$width"), Some(s"$value"))
```

Comment:
My thinking here is that instead of combining components into a raw string and printing that out, the literal `toString` overrides should pass as parameters whatever information they want included as (optional) params to the `stringAccessor`. The `stringAccessor` can then reason about available information and print out the best possible `toString` it has.

But.. I'm still using the `vals` param to `stringAccessor` as a catch-all for all this info - my original plan was to split out the `vals` param into separate chunks (`width`, `type`, `binaryPoint`, ...), but that did not materialize for this PR. So maybe I should revert back to the original design for this PR? We can chat more about this offline